common/metrics/metrics.go

```diff
@@ -140,3 +155,14 @@ type LatencyMetric interface {
 	// when creating the metric, an error will be returned.
 	ReportLatency(latency time.Duration, label ...any)
 }
+
+// RunningAverageMetric tracks the average of a series of values over a given time window.
+type RunningAverageMetric interface {
```

Comment:
You probably don't need this. Running average is a dashboarding problem not metric.

Reply:
Removed `RunningAverageMetric` and replaced with gauges.